aten/src/ATen/native/cuda/BinaryOpsKernel.cu

```diff
@@ -39,8 +40,13 @@ void div_kernel_cuda(TensorIterator& iter) {
       });
     });
   } else {
+    auto assert_state = at::cuda::getCurrentCUDAStream().assert_state();
     AT_DISPATCH_ALL_TYPES_AND(kHalf, iter.dtype(), "div_cuda", [&]() {
-      gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
+      gpu_kernel_with_scalars(iter, [=]GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
+        if (b == 0) {
+            graceful_devision_by_zero(assert_state, "ZeroDivisionError: integer division by zero");
```

Comment:
I don't think this condition is right. `AT_DISPATCH_ALL_TYPES_AND` seems to imply floating point types are also hit here. We shouldn't do the condition test for those kernels.